eea/src/main/java/org/web3j/protocol/eea/crypto/transaction/type/PrivateTransaction1559.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright 2023 Web3 Labs Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package org.web3j.protocol.eea.crypto.transaction.type;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.web3j.crypto.Sign;
+import org.web3j.crypto.transaction.type.ITransaction;
+import org.web3j.rlp.RlpString;
+import org.web3j.rlp.RlpType;
+import org.web3j.utils.Base64String;
+import org.web3j.utils.Bytes;
+import org.web3j.utils.Numeric;
+import org.web3j.utils.Restriction;
+
+public class PrivateTransaction1559 extends RawPrivateTransaction implements ITransaction {
```

Comment:
Is it not a bit odd that RawTransaction does not implement ITransaction, if PrivateTransaction1559 does? It seems like a bit of a broken pattern?

Genuine question: Why should PrivateTransaction1559 implement the interface is RawTransaction does not?